game-core/src/main/java/games/strategy/engine/framework/GameRunner.java

```diff
@@ -109,21 +109,23 @@ public static void main(final String[] args) {
         "UI client launcher invoked from headless environment. This is current prohibited by design to "
             + "avoid UI rendering errors in the headless environment.");
 
-    ErrorHandler.registerExceptionHandler();
     ClientSetting.initialize();
 
     if (!ClientSetting.USE_EXPERIMENTAL_JAVAFX_UI.booleanValue()) {
-      final Console console = new Console();
-      final SimpleFormatter formatter = new SimpleFormatter();
-      LogManager.getLogManager().getLogger("").addHandler(new ConsoleHandler(
-          logMsg -> {
-            if (logMsg.getLevel().intValue() > Level.INFO.intValue()) {
-              ErrorMessage.show(logMsg.getMessage());
-            }
-            console.append(formatter.format(logMsg));
-          }));
-      ErrorMessage.enable();
-      Interruptibles.await(() -> SwingAction.invokeAndWait(LookAndFeel::setupLookAndFeel));
+      Interruptibles.await(() -> SwingAction.invokeAndWait(() -> {
+        LookAndFeel.setupLookAndFeel();
+        final Console console = new Console();
```

Comment:
Are you sure this needs to go on the EDT? AFAIK the components can be created off the EDT, but if rendered they need to be on the EDT. 

I  also wonder about `invokeLater` vs `invokeAndWait`. Would switching to `invokeLater` be better? The call would not be blocking, and googling around on previous cases of crashes, `invokeAndWait` seems to have some pitfalls; I wonder if it's best we avoid using it.

Reply:
I might be dating myself here a bit, I see this recommendation has changed:

https://stackoverflow.com/questions/491323/is-it-safe-to-construct-swing-awt-widgets-not-on-the-event-dispatch-thread

> Sun has changed the rules in 2004 -- before, you were allowed to create the components outside the EDT and only had to move into the EDT once the component had been realized.



> The ErrorHandler#registerExceptionHandler() method contains Swing-specific code and should not be run if the JFX client is running

Okay, but (previously)[https://github.com/triplea-game/triplea/blob/master/game-core/src/main/java/games/strategy/engine/framework/GameRunner.java#L112]   `ErrorHandler#registerExceptionHandler()` was not running on the EDT before this update. IThe swing specific code are magic properties, I'd be interested if I missed the swing specific parts. Unless I'm really too much of dinosaur, strictly minimizing what is run on EDT was a good practice.

Reply:
re: invokeAndWait vs invokeLater, it's not about optimization; invokeAndWait is a bit more hazardous and can more easily lead to lock-up problems:
- https://bugs.eclipse.org/bugs/show_bug.cgi?id=526523
- https://answers.yahoo.com/question/index?qid=20140624231939AAdFii0

I'm not positive if it's necessarily a best practice to avoid invokeAndWait, but it did seem while looking for places that would lock up the EDT `invokeAndWait` locations were suspects over `invokeLater`. That is somewhat anecdotal, I'd be interested to learn further if we have taken steps to make sure `invokeAndWait` is not hazardous for us and that it's not presenting a set of landmines.

Regardless, changing that from invokeAndWait to invokeLater is not necessarily appropriate for this PR. At the same time, correcting/changing which components are touched on which threads is not necessarily either too : )   That is largely why I brought it up. Correcting that swing components are created on EDT is good IMO, I support that.

Reply:
> Okay, but previously `ErrorHandler#registerExceptionHandler()` was not running on the EDT before this update.

Prior to #3715, it _was_ called on the EDT from `ErrorConsole#createConsole()`:

```java
public static void createConsole() {
  Interruptibles.await(() -> SwingAction.invokeAndWait(() -> {
    console = new ErrorConsole();
    console.displayStandardOutput();
    console.displayStandardError();
    ErrorHandler.registerExceptionHandler();
  }));
}
```

> At the same time, correcting/changing which components are touched on which threads is not necessarily either too

It is required as part of this PR to move creation of the console and error dialog to the EDT.  The purpose of this PR is to ensure these components are painted using the user's selected L&F.  That means component creation must be deferred until after the L&F is set.  If the user has chosen Substance, then creating these components off the EDT will result in an exception, as demonstrated above.

Reply:
> It is required as part of this PR to move creation of the console and error dialog to the EDT. 

Are you sure they will only pick up the L&F updates if created on EDT? 
If that is the case, it would be very interesting and I'm pretty sure we would have that problem in other places.

Reply:
> Are you sure they will only pick up the L&F updates if created on EDT?

Please re-read [this comment](https://github.com/triplea-game/triplea/pull/3750#discussion_r208424166).  Component creation was moved to the EDT simply to avoid a `UiThreadingViolationException` being raised when Substance is the selected L&F.

Reply:
Sorry to have been thick there, thank you for the detailed explanation, on a careful reading it is clear now. It's interesting to learn about that enforcement, one really does get to like Swing in retrospect  : )